orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala

```diff
@@ -53,68 +55,74 @@ object ResourceMgr {
     workflowId: String,
     resourceId: String
   ): Behavior[Msg] = Behaviors.setup { ctx =>
-    ctx.log.info(s"Starting ResourceMgr ${ctx.self}")
-
-    val resourceQuery = new ResourceQuery(workflowId, resourceId)
-    val resourceR = database.sync(resourceQuery.get()).get
-
-    // here we make all invalid input to default 8 hours, the input should do validation before
-    // saving them to DB
-    val terminateAfterDuration: FiniteDuration =
-      try { (resourceR.terminateAfter * 1.hour).asInstanceOf[FiniteDuration] }
-      catch { case e: Exception => 8.hour }
-    val ps = Params(
-      ctx,
-      database,
-      resourceQuery,
-      workflowMgr,
-      workflowId,
-      resourceId,
-      resourceR.name,
-      resourceR.maxAttempt,
-      resourceR.resourceType,
-      resourceR.resourceSpec,
-      terminateAfterDuration
-    )
+    Behaviors.withTimers { timers =>
+      ctx.log.info(s"Starting ResourceMgr ${ctx.self}")
 
-    resourceR.status match {
-      case Status.Pending =>
-        idle(ps)
-
-      case Status.Running =>
-        val resourceInsts = database.sync(resourceQuery.instances())
-        val lastInstOpt = resourceInsts.sortBy(_.instanceAttempt)(Ordering[Int].reverse).headOption
-        val instIdEith = lastInstOpt match {
-          case Some(lastInst) =>
-            if (!Status.isAlive(lastInst.status) && lastInst.instanceAttempt < resourceR.maxAttempt) {
-              Right(lastInst.instanceAttempt + 1)
-            } else if (lastInst.status == Status.Activating || lastInst.status == Status.Running) {
-              Right(lastInst.instanceAttempt)
-            } else {
-              Left(lastInst.status)
-            }
-          case None =>
-            Right(1)
-        }
+      val resourceQuery = new ResourceQuery(workflowId, resourceId)
+      val resourceR = database.sync(resourceQuery.get()).get
 
-        val result = for {
-          instId <- instIdEith
-          rscInst <- spawnResourceInstance(
-            ctx,
-            database,
-            ps,
-            instId
-          )
-        } yield running(ps, rscInst, instId)
+      // here we make all invalid input to default 8 hours, the input should do validation before
+      // saving them to DB
+      val terminateAfterDuration: FiniteDuration =
+      try {
+        (resourceR.terminateAfter * 1.hour).asInstanceOf[FiniteDuration]
+      }
+      catch {
```

Comment:
can you join this with the previous line?

Reply:
done